src/java/org/apache/cassandra/cql3/restrictions/StatementRestrictions.java

```diff
@@ -476,6 +507,38 @@ public boolean usesSecondaryIndexing()
         return this.usesSecondaryIndexing;
     }
 
+    /**
+     * This is a hack to push ordering down to indexes.
+     * Indexes are selected based on RowFilter only, so we need to turn orderings into restrictions
+     * so they end up in the row filter.
+     *
+     * @param orderings orderings from the select statement
+     * @return the {@link RestrictionSet} with the added orderings
+     */
+    private RestrictionSet addOrderingRestrictions(List<Ordering> orderings, RestrictionSet restrictionSet)
+    {
+        List<Ordering> annOrderings = orderings.stream().filter(o -> o.expression.hasNonClusteredOrdering()).collect(Collectors.toList());
+
+        if (annOrderings.size() > 1)
+            throw new InvalidRequestException("Cannot specify more than one ANN ordering");
+        else if (annOrderings.size() == 1)
+        {
+            if (orderings.size() > 1)
+                throw new InvalidRequestException("ANN ordering does not support secondary ordering");
```

Comment:
I would say something like `ANN ordering does not support any other ordering`, since ANN could be considered the secondary ordering.

Reply:
+1

Reply:
I think we are still missing this one in the last batch of commits.